articles/performance-optimization-techniques-for-node-js/index.md

```diff
@@ -0,0 +1,368 @@
+
+Commonly used method for logging in Node.js are `console.log()` which logs Standard outputs, (stdout) and `console.error()` which will log standard errors (stderr). However, there are more efficient, reliable, popular, and scalable libraries/3rd party APIs that will assist you in logging scripts in your Node.js app.
+
+They include:
+
+#### 1. Winston
+[Winston](https://github.com/winstonjs/winston) is a multi-transport async logger, which is simple, universal, and extremely versatile. It makes logging more flexible and extensible. It is the most popular logger based on NPM stats. Winston has different transports with different default levels which indicates message priority. [These levels include](https://github.com/winstonjs/winston#logging-levels)
+
+   - error
+   - warn
+   - info
+   - http
+   - verbose
+   - debug
+   - silly
+
+[**Example of a simple/custom Winston logger**](https://github.com/winstonjs/winston#creating-your-own-logger)
+
+```js
+const winston = require("winston");
+const logger = winston.createLogger({
+  transports: [new winston.transports.Console()],
+});
+logger.info("Information message");
+logger.warn("Warning message");
+logger.error("Error message");
+```
+[**Winston logger with file transport**](https://github.com/winstonjs/winston#multiple-transports-of-the-same-type)
+```js
+const winston = require("winston");
+// Logger configuration
+const logConfiguration = {
+  transports: [
+    new winston.transports.Console({
+      level: "verbose",
+    }),
+    new winston.transports.File({
+      level: "error",
+      filename: "./logs/example-3.log",
+    }),
+  ],
+};
+//Create logging configurations
+const logger = winston.createLogger(logConfiguration);
+// Log some messages
+logger.silly("Trace message, Winston!");
+logger.debug("Debug message, Winston!");
+logger.verbose("A bit more info, Winston!");
+logger.info("Hello, Winston!");
+logger.warn("Heads up, Winston!");
+logger.error("Danger, Winston!");
+```
+#### 2. Morgan
+
+[Morgan](https://github.com/expressjs/morgan) is an HTTP request logger middleware for Node.js applications. Morgan gives insights on how your app is being used and alerts you on potential errors and issues that could be threats to your application. Morgan is considered the most reliable HTTP logger by Node.js developers. Morgan is designed to log errors the way servers like [Apache](https://httpd.apache.org/docs/1.3/logs.html) and [Nginx](https://docs.nginx.com/nginx/admin-guide/monitoring/logging/) do to the access-log or error-log.
+
+Sample app that will log all requests in the [Apache](https://httpd.apache.org/docs/1.3/logs.html) combined format to one log file per day in the log/ directory using the [rotating-file-stream module](https://www.npmjs.com/package/rotating-file-stream).
```

Comment:
"Sample app that will log all requests in the [Apache](https://httpd.apache.org/docs/1.3/logs.html) combined format to one log file per day in the log/ directory using the [rotating-file-stream module](https://www.npmjs.com/package/rotating-file-stream)." 

This sentence was confusing me a bit - is it an example ??

Reply:
Yeah, @hectorkambow . Goes with the code snippet below